service/controller/clusterapi/v17/key/cluster_provider.go

```diff
@@ -41,6 +41,30 @@ func g8sClusterSpecFromCMAClusterSpec(cmaSpec cmav1alpha1.ProviderSpec) (g8sv1al
 	return g8sSpec, nil
 }
 
+func g8sClusterCommonStatusFromCMAClusterStatus(cmaStatus *runtime.RawExtension) g8sv1alpha1.CommonClusterStatus {
+	type wrapper struct {
+		Cluster g8sv1alpha1.CommonClusterStatus `json:"cluster" yaml:"cluster"`
+	}
```

Comment:
The implementation of the other methods here is provider specific which as we discussed are a bit off already. For the collector I wanted to have something more generic. Whatever provider status we unmarshal here, the wrapper should extract the common cluster status types that way.

Reply:
This is actually a very clever way to deal with this. :+1:

Reply:
> Whatever provider status we unmarshal here, the wrapper should extract the common cluster status types that way.

I'd like to have that as a comment here.

Reply:
Added the comment.